rust/processor/src/processors/parquet_processors/parquet_ans_processor.rs

```diff
@@ -0,0 +1,263 @@
+        ans_lookup::CurrentAnsPrimaryName,
+        parquet_ans_lookup_v2::{AnsPrimaryNameV2, CurrentAnsPrimaryNameV2},
+    },
+    gap_detectors::ProcessingResult,
+    processors::{ProcessorName, ProcessorTrait},
+    utils::{counters::PROCESSOR_UNKNOWN_TYPE_COUNT, database::ArcDbPool},
+};
+use ahash::AHashMap;
+use anyhow::anyhow;
+use aptos_protos::transaction::v1::{
+    transaction::TxnData, write_set_change::Change as WriteSetChange, Transaction,
+};
+use async_trait::async_trait;
+use kanal::AsyncSender;
+use serde::{Deserialize, Serialize};
+use std::{fmt::Debug, time::Duration};
+use tracing::error;
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct ParquetAnsProcessorConfig {
+    pub google_application_credentials: Option<String>,
+    pub bucket_name: String,
+    pub bucket_root: String,
+    pub parquet_handler_response_channel_size: usize,
+    pub max_buffer_size: usize,
+    pub ans_v1_primary_names_table_handle: String,
+    pub ans_v1_name_records_table_handle: String,
+    pub ans_v2_contract_address: String,
+    pub parquet_upload_interval: u64,
+}
+
+impl UploadIntervalConfig for ParquetAnsProcessorConfig {
+    fn parquet_upload_interval_in_secs(&self) -> Duration {
+        Duration::from_secs(self.parquet_upload_interval)
+    }
+}
+
+pub struct ParquetAnsProcessor {
+    connection_pool: ArcDbPool,
+    config: ParquetAnsProcessorConfig,
+    ans_primary_name_v2_sender: AsyncSender<ParquetDataGeneric<AnsPrimaryNameV2>>,
+}
+
+impl ParquetAnsProcessor {
+    pub fn new(
+        connection_pool: ArcDbPool,
+        config: ParquetAnsProcessorConfig,
+        new_gap_detector_sender: AsyncSender<ProcessingResult>,
+    ) -> Self {
+        if let Some(credentials) = config.google_application_credentials.clone() {
+            std::env::set_var(GOOGLE_APPLICATION_CREDENTIALS, credentials);
+        }
+
+        let ans_primary_name_v2_sender = create_parquet_handler_loop::<AnsPrimaryNameV2>(
+            new_gap_detector_sender.clone(),
+            ProcessorName::ParquetAnsProcessor.into(),
+            config.bucket_name.clone(),
+            config.bucket_root.clone(),
+            config.parquet_handler_response_channel_size,
+            config.max_buffer_size,
+            config.parquet_upload_interval_in_secs(),
+        );
+
+        Self {
+            connection_pool,
+            config,
+            ans_primary_name_v2_sender,
+        }
+    }
+}
+
+impl Debug for ParquetAnsProcessor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let state = &self.connection_pool.state();
+        write!(
+            f,
+            "AnsProcessor {{ connections: {:?}  idle_connections: {:?} }}",
+            state.connections, state.idle_connections
+        )
+    }
+}
+
+#[async_trait]
+impl ProcessorTrait for ParquetAnsProcessor {
+    fn name(&self) -> &'static str {
+        ProcessorName::AnsProcessor.into()
```

Comment:
do we want a processor name that can differentiate parquet and regular ones? I see this method is implemented in different ways: 1) from processor name 2) hard coding

Reply:
good catch! will fix this.